### PR TITLE
Provide resizing ratios option

### PR DESCRIPTION
**Pull request by Bladrak**

### CODE REVIEW

File: src/js/components/Resize.jsx [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
 
-
 var Resize = React.createClass({
     mixins: [Reflux.ListenerMixin],
 
     render: function () {
+        var select = "";
+
+        if (_.isArray(config.presetsResize)) {
+            this.props.presets = [{
```
Comment (plouc) [inline src/js/components/Resize.jsx]: @Bladrak the component must not populate its own props

Comment (plouc) [inline src/js/components/Resize.jsx]: If you want to pre-populate it, you should do it from the parent component

**Comment by Bladrak** [comment]
@plouc updated

File: src/js/App.jsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ React.render((
             {presetSelector}
             {serverSelector}
             <ImageSource />
-            <Resize />
+            <Resize presets={config.presetsResize} />
```
Comment (plouc) [inline src/js/App.jsx]: Should be
```javascript
config.presetResize || []
```

File: src/js/components/Resize.jsx [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
 var ResizeActions = require('./../actions/ResizeActions');
 
-
 var Resize = React.createClass({
     mixins: [Reflux.ListenerMixin],
 
     render: function () {
+        var select = "";
+
+        if (_.isArray(this.props.presets)) {
```
Comment (plouc) [inline src/js/components/Resize.jsx]: You should use http://facebook.github.io/react/docs/reusable-components.html#prop-validation

Comment (plouc) [inline src/js/components/Resize.jsx]: You should also check the length of the presets array

**Comment by Bladrak** [comment]
@plouc updated

File: src/js/components/Resize.jsx [hunk 4]
```diff
@@ -1,12 +1,39 @@
...
 var React         = require('react/addons');
 var Reflux        = require('reflux');
 var ResizeActions = require('./../actions/ResizeActions');
 
-
 var Resize = React.createClass({
     mixins: [Reflux.ListenerMixin],
 
+    props: {
+        presets: React.PropTypes.array,
```
Comment (plouc) [inline src/js/components/Resize.jsx]: Should be optional

**Pull request merged**